### PR TITLE
Fixed the issue #474

### PR DESCRIPTION
**Pull request by heidijinxujia**
1. fixed the issue #474
    Add the feature to ensure that all redirects that happen in the web view are SSL protected (start with "https://") to prevent phishing attacks.
2. renamed some constants
3. added some log
4. added the unit test for the redirect URL checking (in progress, more corner cases need to test)

### CODE REVIEW

**Comment by heidijinxujia** [comment]
@RPangrle

File: src/src/com/microsoft/aad/adal/ADALError.java [hunk 1]
```diff
@@ -466,6 +466,11 @@
      * WebView returned empty redirect url.
      */
     WEBVIEW_RETURNED_EMPTY_REDIRECT_URL("Webview returned empty redirect url"),
+    
+    /**
+     * WebView  redirect url is not SSL protected.
+     */
+    WEBVIEW_REDIRECTURL_NOT_SSL_PROTECTED("Webview redirect url is not SSL protected"),
```
Comment (RPangrle) [inline src/src/com/microsoft/aad/adal/ADALError.java]: I suspect this would be too easily confused with the "redirect uri" parameter, something the developer has control over. In this case the issue is with the server configuration. "The webview was redirected to an unsafe URL" would be a little more clear on that point.

**Comment by heidijinxujia** [comment]
@RPangrle Fixed, how about the rest code? Thanks!

**Comment by heidijinxujia** [comment]
@weijjia

File: src/src/com/microsoft/aad/adal/AuthenticationConstants.java [hunk 2]
```diff
@@ -303,7 +303,7 @@
         
         public static final String AZURE_AUTHENTICATOR_APP_PACKAGE_NAME = "com.azure.authenticator";
 
-        public static final String PKEYAUTH_REDIRECT = "urn:http-auth:PKeyAuth";
+        public static final String CLIENT_TLS_REDIRECT = "urn:http-auth:PKeyAuth";
```
Comment (weijjia) [inline src/src/com/microsoft/aad/adal/AuthenticationConstants.java]: why have the name changed?

Comment (heidijinxujia) [inline src/src/com/microsoft/aad/adal/AuthenticationConstants.java]: You asked me to change it last time. :)

Comment (weijjia) [inline src/src/com/microsoft/aad/adal/AuthenticationConstants.java]: no.. i don't remember sorry. I think I changed it to the pkeyauth_redirect in the previous PRs. it's actually pkeyauth_redirect, not client_tls_redirect.

File: src/src/com/microsoft/aad/adal/BasicWebViewClient.java [hunk 3]
```diff
@@ -148,11 +148,14 @@ public void onPageStarted(WebView view, String url, Bitmap favicon) {
         showSpinner(true);
     }
     
+
```
Comment (weijjia) [inline src/src/com/microsoft/aad/adal/BasicWebViewClient.java]: nitpick: extra lines.

File: src/src/com/microsoft/aad/adal/AuthenticationActivity.java [hunk 4]
```diff
@@ -608,14 +609,28 @@ public void processRedirectUrl(final WebView view, String url){
         }
         
         public boolean processInvalidUrl(final WebView view, String url) {
+        	final String methodName = ".processInvalidUrl";
```
Comment (weijjia) [inline src/src/com/microsoft/aad/adal/AuthenticationActivity.java]: nitpick: can you change the . to :? The whole code I think it's using :, we should make it consistent. If we want to change it later, then it should be across file changes.

File: src/src/com/microsoft/aad/adal/AuthenticationConstants.java (hunk 2 above)
Comment (heidijinxujia) [inline src/src/com/microsoft/aad/adal/AuthenticationConstants.java]: Alright, I will change it back. :)

**Comment by weijjia** [comment]
:shipit:
